pubsub/google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -153,8 +153,12 @@ def drop(self, items):
 
         Args:
             items(Sequence[DropRequest]): The items to drop.
+            activate_ordering_keys(bool): Whether or not to send the next message
+                for ordered messages in `items`, if they exist.
         """
         self._manager.leaser.remove(items)
+        ordering_keys = (k.ordering_key for k in items if k.ordering_key)
+        self._manager.activate_ordering_keys(ordering_keys)
```

Comment:
This might be later than it needs to be for delivering the next message for the ordering key. We only need to wait for the user callback to run to completion, we do not need to wait for ack or nack. Maybe in Python it doesn't matter so much because the likelihood of someone doing asynchronous processing outside of the callback is small.

Reply:
My assumption was that most users would ack their message inside the user callback itself. This would be before the callback finishes, though presumably near the end. I think this scenario is more likely than offloading the message to an async process. We can revisit the question if the async scenario arises in practice. The change itself would be small - modifying _wrap_callback_errors in streaming_pull_manager.py.